syft/frameworks/torch/tensors/interpreters/gradients_core.py

```diff
@@ -38,6 +46,70 @@ def __call__(self, grad):
     def __repr__(self):
         return self.__class__.__name__
 
+    def __setattr__(self, name, value):
+        self.__dict__[name] = value
+
+        # add attributes of grad function to _parameters list
+        # essential for serialization and deserialization
+        if not name in ["next_functions", "result", "_parameters"]:
+            self._parameters.append(value)
+
+    @staticmethod
+    def simplify(worker: AbstractWorker, grad_fn) -> tuple:
+        """ Takes the attributes of a grad_fn object and saves them in a tuple
+            Every gradient function class that extends `GradFunc` uses this function to
+            simplify the attributes.
+
+        Args:
+            grad_fn: gradient function object (AddBackward, SubBackward, etc.)
+
+        Returns:
+            grad_fn_params: a tuple containing all the simplified attributes
+            of gradient function
+        """
+
+        # Add class name of the grad_fn object at the begining.
+        # Essential while deserializing the object
+        cls = syft.serde.msgpack.serde._simplify(worker, grad_fn.__class__.__name__)
+        grad_fn_params = [cls]
+
+        for param in grad_fn._parameters:
+
+            if hasattr(param, "child"):
+                param = syft.serde.msgpack.serde._simplify(worker, param.child)
+            else:
+                param = syft.serde.msgpack.serde._simplify(worker, param)
+
+            grad_fn_params.append(param)
+
+        return tuple(grad_fn_params)
+
+    @staticmethod
+    def detail(worker: AbstractWorker, gradfn_tuple):
+        """ This function reconstructs (deserializes) the gradient function object,
+         given its attributes in the form of a tuple
+
+         Args:
+            gradfn_tuple: a tuple containing all the simplified attributes of a
+            grad function
+
+        Returns:
+            A correct gradient function object
+
+        """
+        # Extract the class name from tuple
+        cls = syft.serde.msgpack.serde._detail(worker, gradfn_tuple[0])
```

Comment:
I think that you can all `syft.serde.msgpack.serde._detail` directly on the list, which would make your syntax lighter:
smthg like this
`cls_name, *grad_fn_params = syft.serde.msgpack.serde._detail(worker, gradfn_tuple)`

Reply:
Hey, I have changed the functions. Please review :)